Sample Apps/Storefront/Storefront/AddressViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  AddressViewModel.swift
+//  Storefront
+//
+//  Created by Shopify.
+//  Copyright (c) 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import Buy
+
+final class AddressViewModel: ViewModel {
```

Comment:
While it's true that oftentimes a view model represents a dumb immutable container capable of nothing more than holding values but in this case, collection and product view models represent identity and not value. In the hybrid collection layout we have `CollectionViewModel` types that each have `products` represented by a `PageableArray<ProductViewModel>`. Each pageable array can then be individually paginated and represents the current state paging state (offset, limit, etc). If these view models were value types, updating the contents of each pageable array would involve setting new values on all instances of the collection view model. If, for example, there's more than one view controller presenting the same model on screen, the logic for finding and updating that collection would have to exist in each view controller separately. Having view models as reference types means that we can update products in a single place and have the changes propagate throughout. We want shared mutable state in this case.